recipes/shaderc/all/conanfile.py

```diff
@@ -0,0 +1,105 @@
+import os
+
+from conans import ConanFile, CMake, tools
+
+class ShadercConan(ConanFile):
+    name = "shaderc"
+    description = "A collection of tools, libraries and tests for shader compilation."
+    license = "Apache-2.0"
+    topics = ("conan", "shaderc", "glsl", "hlsl", "msl", "spirv", "spir-v", "glslc", "spvc")
+    homepage = "https://github.com/google/shaderc"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "spvc": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "spvc": False
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 11)
+
+    def requirements(self):
+        self.requires("glslang/8.13.3559")
+        self.requires("spirv-tools/v2020.3")
+        if self.options.spvc:
+           self.requires("spirv-cross/20200519")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-" + self.version, self._source_subfolder)
+
+    def build(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["CMAKE_POSITION_INDEPENDENT_CODE"] = self.options.get_safe("fPIC", True)
```

Comment:
```suggestion
```

Not required, it's automatically configured by CMake helper: https://docs.conan.io/en/latest/reference/build_helpers/cmake.html#definitions

Reply:
It's required in shaderc recipe, because shaderc_util is always static and linked into shaderc. conan does not inject PIC when option is removed, leading to default behaviour (non PIC for static, which is wrong setting for shaderc_util if shaderc shared).

Reply:
Thanks for clarifying!